crates/jstzd/src/protocol/bootstrap.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::HashSet;
+
+use jstz_crypto::public_key::PublicKey;
+use serde_json::Value;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct BootstrapAccount {
+    public_key: PublicKey,
+    amount_mutez: u64,
+}
+
+impl BootstrapAccount {
+    pub fn get_amount(&self) -> u64 {
+        self.amount_mutez
+    }
+
+    pub fn get_public_key(&self) -> &PublicKey {
+        &self.public_key
+    }
+}
+
+impl TryFrom<&Value> for BootstrapAccount {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &Value) -> Result<Self, Self::Error> {
+        let array = value
+            .as_array()
+            .ok_or(anyhow::anyhow!("value is not a valid json array"))?;
```

Comment:
I wanted to improve the error messages a bit, but I think I'll do that later after all these major changes are done.